chia/types/borderlands.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from typing import NewType
+
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes48, bytes96
+
+PrivateKeyBytes = NewType("PrivateKeyBytes", bytes32)
+PublicKeyBytes = NewType("PublicKeyBytes", bytes48)
+SignatureBytes = NewType("SignatureBytes", bytes96)
+PuzzleHash = NewType("PuzzleHash", bytes32)
+CoinID = NewType("CoinID", bytes32)
+SpendBundleID = NewType("SpendBundleID", bytes32)  # Also used as MempoolItemID
+BlockRecordHeaderHash = NewType("BlockRecordHeaderHash", bytes32)
+
+
+def bytes_to_PrivateKeyBytes(input_bytes: bytes) -> PrivateKeyBytes:
+    return PrivateKeyBytes(bytes32(input_bytes))
+
+
+def bytes_to_PublicKeyBytes(input_bytes: bytes) -> PublicKeyBytes:
```

Comment:
`bytes_to_PrivateKeyBytes` etc are needed to avoid importing `bytes32` and then writing `PrivateKeyBytes(bytes32(foo))`.

Reply:
I would hope there would be a way to "convert" types here without actually cloning/re-allocating the bytes in memory. Do you know whether this has a run-time cost or if it's just a mypy check conversion?

Reply:
I think there is a small runtime cost, but I don't know if the inner object is wrapped at runtime.
I don't think it is, because the checks are definitely not performed at runtime. (The typecheck-time and runtime "instantiations" are totally different implementations)